src/serialization/interval_log/mod.rs

```diff
@@ -0,0 +1,667 @@
+//! assert_eq!("#Comments are great\n", &str::from_utf8(&buf).unwrap()[0..20]);
+//! ```
+
+extern crate base64;
+
+use std::{fmt, io, ops, str, time};
+use std::fmt::Write;
+
+use nom::{double, line_ending, not_line_ending, IResult};
+
+use super::super::{Counter, Histogram};
+use super::Serializer;
+
+/// Prepare an `IntervalLogWriter`.
+///
+/// This type only allows writing comments and headers. Once you're done writing those things, use
+/// `into_log_writer()` to convert this into an `IntervalLogWriter`.
+pub struct IntervalLogWriterBuilder {
+    comments: Vec<String>,
+    start_time: Option<f64>,
+    base_time: Option<f64>,
+    max_value_divisor: f64,
+}
+
+impl IntervalLogWriterBuilder {
+    /// Create a new log writer that writes to `writer` and serializes histograms with `serializer`.
+    pub fn new() -> IntervalLogWriterBuilder {
+        IntervalLogWriterBuilder {
+            comments: Vec::new(),
+            start_time: None,
+            base_time: None,
+            max_value_divisor: 1.0,
+        }
+    }
+
+    /// Add a comment line to be written when the writer is built.
+    ///
+    /// If you do silly things like write a comment with a newline character, you'll end up with
+    /// an un-parseable log file. Don't do that.
+    pub fn add_comment(&mut self, s: &str) -> &mut Self {
+        self.comments.push(s.to_owned());
+        self
+    }
+
+    /// Set a StartTime. See the module-level documentation for more info.
+    ///
+    /// This can be called multiple times, but only the value for the most recent invocation will
+    /// be written.
+    pub fn with_start_time(&mut self, seconds_since_epoch: f64) -> &mut Self {
+        self.start_time = Some(seconds_since_epoch);
+        self
+    }
+
+    /// Set a BaseTime. See the module-level documentation for more info.
+    ///
+    /// This can be called multiple times, but only the value for the most recent invocation will
+    /// be written.
+    pub fn with_base_time(&mut self, seconds_since_epoch: f64) -> &mut Self {
+        self.base_time = Some(seconds_since_epoch);
+        self
+    }
+
+    /// Set a max value divisor.
+    ///
+    /// This is used to scale down the max value part of an interval log to something that may be
+    /// more human readable. The max value in the log is only for human consumption, so you might
+    /// prefer to divide by 10<sup>9</sup> to turn nanoseconds into fractional seconds, for
+    /// instance.
+    ///
+    /// If this is not set, 1.0 will be used.
+    ///
+    /// This can be called multiple times, but only the value for the most recent invocation will
+    /// be written.
+    pub fn with_max_value_divisor(&mut self, max_value_divisor: f64) -> &mut Self {
+        self.max_value_divisor = max_value_divisor;
+        self
+    }
+
+    /// Build a LogWriter and apply any configured headers.
+    pub fn build_with<'a, 'b, W: 'a + io::Write, S: 'b + Serializer>(
```

Comment:
This is pretty much the opposite of what https://aturon.github.io/ownership/builders.html suggests: we're taking the required things at the end, not the beginning, but being able to use the same setup on multiple writers or serializers seems hypothetically useful, and this API does not offend me otherwise.

Let the name bikeshedding begin! :P

Reply:
Eh, I'm okay with this approach.

I actually don't mind `build_with`. I think I'd be partial to a name that fits better in context though. How about `begin_log_with`?

Reply:
Sure, WFM.